lib/esConnector.js

```diff
@@ -343,9 +344,22 @@ ESConnector.prototype.addDefaults = function (modelName) {
     }
     filter.type = typeFromDatasource || this.searchType || modelName;
 
+    // When changing data, wait until the change has been indexed so it is instantly available for search
+    if(['create', 'save', 'destroy', 'destroyAll', 'updateAttributes', 'updateOrCreate'].indexOf(functionName) != -1) {
```

Comment:
@wolfgang-s I want this array to come from datasource.json file. User should enter the methods names in an array and pass it on to connector. If no array is present then as a fallback use this array and make refresh true, else whatever methods users sends through array they will only have refresh true.

You can access the array from datasource file [in this way](https://github.com/strongloop-community/loopback-connector-elastic-search/blob/master/lib/esConnector.js#L45)

Something like this:
In datasource file:

```
"es": {
        "name": "es",
        "refreshOn": ["save","create", "updateOrCreate"],
        .....
```
and in esConnector.js
```
if (self.refreshOn.indexOf(functionName) != -1) {
        filter.refresh = true;
    }
```